app/validators/poll_expiration_validator.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+class PollExpirationValidator < ActiveModel::Validator
+  MAX_EXPIRATION = 1.month.freeze
+  MIN_EXPIRATION = 5.minutes.freeze
+
+  def validate(poll)
+    current_time = Time.now.utc
+
+    poll.errors.add(:expires_at, I18n.t('polls.errors.duration_too_long')) if poll.expires_at.nil? || poll.expires_at - current_time > MAX_EXPIRATION
```

Comment:
I realize this was just moving around existing code, not something changing here, but...

- In the parent class we have a presence validation on `expires_at` which runs when poll is local
- This validator class runs on local and expires_at_changed?

It seems sort of odd to add a "too long" error when the value is missing, given that it will already have the presence validation error added. I wonder if we could have this validator only run when expires_at is present (and keep the changed, and poll is local checks as-is) - which would let this validator remove those extra checks, assume it has a value, and just do the too_long/too_short checks?